flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/HiveScalarFunction.java

```diff
@@ -82,6 +84,13 @@ public void open(FunctionContext context) {
 		isArgsSingleArray = HiveFunctionUtil.isSingleBoxedArray(argTypes);
 	}
 
+	@Override
+	public TypeInference getTypeInference(DataTypeFactory typeFactory) {
+		TypeInference.Builder builder = TypeInference.newBuilder();
+		builder.outputTypeStrategy(new ResultTypeStrategy(this));
```

Comment:
I guess we will also need a matching input type strategy such that the validation can work properly.